packages/universe-components/src/donationForm/DonationForm.tsx

```diff
@@ -153,12 +197,78 @@ export default function DonationForm({ initiative }: DonationFormProps) {
       setLoading(true);
       setButtonMessage('Approving payment...');
 
-      const paymentResult = await sendPayment(destinationWalletAddress, amount);
+      console.log('FORM', donationForm)
+      console.log('CHAIN', chainState)
+      const amountToSend = coinAmount
+      //const amountToSend = 0.0001
+      //const amountToSend:number = donationForm.showUsd ? (usdAmount / chainState.exchangeRate) : coinAmount;
+      //console.log('AMTS', donationForm.showUsd, amount, usdAmount, coinAmount, chainState.exchangeRate)
+      console.log('SEND', amountToSend)
+      const paymentResult = await sendPayment(destinationWalletAddress, amountToSend);
 
       if (!paymentResult.success) {
         throw new Error(`Payment error: ${paymentResult.error ?? 'unknown'}`);
       }
 
+      const network = appConfig.chainDefaults.network || 'testnet'
+      const chainName = chainInterface?.chain.name || ''
+      const currency = chainInterface?.chain.symbol || ''
+      const sender = paymentResult.walletAddress || ''
+      
+      // Save user if not exists
+      //const user = (await getUserByWallet(sender)) || ({} as UserRecord);
+      const user = await fetchUserByWallet(sender)
+      console.log('USER', user)
+      let userId = user?.id || ''
+      let userKey = user?.api_key || ''
+      if(!userId){
+        const anon = await createAnonymousUser({walletAddress:sender, chain:chainName, network, tba:true})
+        console.log('ANON', anon)
+        if(!anon){
+          console.log('Error creating anonymous user')
+          setButtonMessage('Error saving user data, contact support')
+          return false
+        }
+        userId = anon.id
+        userKey = anon.api_key || ''
+      }
+
+      // Save donation to DB
+      const donationData = {
+        organizationId: organization.id, 
+        initiativeId: initiative.id, 
+        categoryId: initiative?.categoryId || organization?.categoryId || '',
+        userId,
+        sender,
+        chainName,
+        network, 
+        currency,
+        coinValue: coinAmount, 
+        usdValue: usdAmount, 
+      }
```

Comment:
```suggestion
      const donationData = {
        organization: {
          connect: { id: organization.id}
        }, 
        initiative: {
          connect: { id: initiative.id },
        }, 
        category: {
          connect: {id: initiative?.categoryId || organization?.categoryId || ''}
        },
        userId,
        sender,
        chainName,
        network, 
        currency,
        coinValue: coinAmount, 
        usdValue: usdAmount, 
      }
```

Reply:
Is that the way we solve the previous issue? Whowuldathunk!